### PR TITLE
fix: FFI/cgo string passing

### LINKED ISSUE #131: Installation of 3rd party helm charts fails sporadically
### Affected version

1.0.0-rc2

### Current and expected behavior

When installing using stackablectl I encounter frequent failures with the installation of helm charts. With debug enabled I see the message, error converting JSON to YAML: yaml: control characters are not allowed. A repeat run of stackablectl sometimes fixes this, but it frequently takes several runs before it will work. The issue appears to occur for all helm charts, including those I've hosted on a local web server in an attempt to rule out 3rd party services as being the cause.

### Possible solution

_No response_

### Additional context

Below are the logs from a failed run where the affected helm chart was Vector, but I have also seen this same issue with OpenSearch and Bitnami PostgreSQL.

```
  DEBUG  Installing manifest from Helm chart ./stacks/stackable-poc/vector-aggregator.yaml
    at rust/stackable-cockpit/src/platform/stack/spec.rs:235
    in install_manifests
    in install_stack_manifests
    in install_cmd with args: DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }, common_args: Cli { log_level: Some(Level(Debug)), no_cache: false, offline: false, files: CommonFileArgs { demo_files: ["demos/demos-v2.yaml"], stack_files: ["stacks/stacks-v2.yaml"], release_files: [] }, repos: CommonRepoArgs { helm_repo_stable: "https://repo.stackable.tech/repository/helm-stable/", helm_repo_test: "https://repo.stackable.tech/repository/helm-test/", helm_repo_dev: "https://repo.stackable.tech/repository/helm-dev/" }, subcommand: Demo(DemoArgs { subcommand: Install(DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }) }) }, transfer_client: FileTransferClient { client: Client { accepts: Accepts, proxies: [Proxy(System({}), None)], referer: true, default_headers: {"accept": "*/*"} }, cache: Cache { auto_purge_interval: 900s, backend: Disk { base_path: "/home/jim/.cache/stackablectl" }, max_age: 3600s } }
    in run with self: DemoArgs { subcommand: Install(DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }) }, common_args: Cli { log_level: Some(Level(Debug)), no_cache: false, offline: false, files: CommonFileArgs { demo_files: ["demos/demos-v2.yaml"], stack_files: ["stacks/stacks-v2.yaml"], release_files: [] }, repos: CommonRepoArgs { helm_repo_stable: "https://repo.stackable.tech/repository/helm-stable/", helm_repo_test: "https://repo.stackable.tech/repository/helm-test/", helm_repo_dev: "https://repo.stackable.tech/repository/helm-dev/" }, subcommand: Demo(DemoArgs { subcommand: Install(DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }) }) }, cache: Cache { auto_purge_interval: 900s, backend: Disk { base_path: "/home/jim/.cache/stackablectl" }, max_age: 3600s }
    in run with self: Cli { log_level: Some(Level(Debug)), no_cache: false, offline: false, files: CommonFileArgs { demo_files: ["demos/demos-v2.yaml"], stack_files: ["stacks/stacks-v2.yaml"], release_files: [] }, repos: CommonRepoArgs { helm_repo_stable: "https://repo.stackable.tech/repository/helm-stable/", helm_repo_test: "https://repo.stackable.tech/repository/helm-test/", helm_repo_dev: "https://repo.stackable.tech/repository/helm-dev/" }, subcommand: Demo(DemoArgs { subcommand: Install(DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }) }) }

   INFO  Installing Helm chart vector (0.26.0)
    at rust/stackable-cockpit/src/platform/stack/spec.rs:244
    in install_manifests
    in install_stack_manifests
    in install_cmd with args: DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }, common_args: Cli { log_level: Some(Level(Debug)), no_cache: false, offline: false, files: CommonFileArgs { demo_files: ["demos/demos-v2.yaml"], stack_files: ["stacks/stacks-v2.yaml"], release_files: [] }, repos: CommonRepoArgs { helm_repo_stable: "https://repo.stackable.tech/repository/helm-stable/", helm_repo_test: "https://repo.stackable.tech/repository/helm-test/", helm_repo_dev: "https://repo.stackable.tech/repository/helm-dev/" }, subcommand: Demo(DemoArgs { subcommand: Install(DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }) }) }, transfer_client: FileTransferClient { client: Client { accepts: Accepts, proxies: [Proxy(System({}), None)], referer: true, default_headers: {"accept": "*/*"} }, cache: Cache { auto_purge_interval: 900s, backend: Disk { base_path: "/home/jim/.cache/stackablectl" }, max_age: 3600s } }
    in run with self: DemoArgs { subcommand: Install(DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }) }, common_args: Cli { log_level: Some(Level(Debug)), no_cache: false, offline: false, files: CommonFileArgs { demo_files: ["demos/demos-v2.yaml"], stack_files: ["stacks/stacks-v2.yaml"], release_files: [] }, repos: CommonRepoArgs { helm_repo_stable: "https://repo.stackable.tech/repository/helm-stable/", helm_repo_test: "https://repo.stackable.tech/repository/helm-test/", helm_repo_dev: "https://repo.stackable.tech/repository/helm-dev/" }, subcommand: Demo(DemoArgs { subcommand: Install(DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }) }) }, cache: Cache { auto_purge_interval: 900s, backend: Disk { base_path: "/home/jim/.cache/stackablectl" }, max_age: 3600s }
    in run with self: Cli { log_level: Some(Level(Debug)), no_cache: false, offline: false, files: CommonFileArgs { demo_files: ["demos/demos-v2.yaml"], stack_files: ["stacks/stacks-v2.yaml"], release_files: [] }, repos: CommonRepoArgs { helm_repo_stable: "https://repo.stackable.tech/repository/helm-stable/", helm_repo_test: "https://repo.stackable.tech/repository/helm-test/", helm_repo_dev: "https://repo.stackable.tech/repository/helm-dev/" }, subcommand: Demo(DemoArgs { subcommand: Install(DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }) }) }

  DEBUG  Add Helm repo
    at rust/stackable-cockpit/src/helm.rs:394
    in add_repo with repo_name: "vector", repo_url: "https://helm.vector.dev"
    in install_manifests
    in install_stack_manifests
    in install_cmd with args: DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }, common_args: Cli { log_level: Some(Level(Debug)), no_cache: false, offline: false, files: CommonFileArgs { demo_files: ["demos/demos-v2.yaml"], stack_files: ["stacks/stacks-v2.yaml"], release_files: [] }, repos: CommonRepoArgs { helm_repo_stable: "https://repo.stackable.tech/repository/helm-stable/", helm_repo_test: "https://repo.stackable.tech/repository/helm-test/", helm_repo_dev: "https://repo.stackable.tech/repository/helm-dev/" }, subcommand: Demo(DemoArgs { subcommand: Install(DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }) }) }, transfer_client: FileTransferClient { client: Client { accepts: Accepts, proxies: [Proxy(System({}), None)], referer: true, default_headers: {"accept": "*/*"} }, cache: Cache { auto_purge_interval: 900s, backend: Disk { base_path: "/home/jim/.cache/stackablectl" }, max_age: 3600s } }
    in run with self: DemoArgs { subcommand: Install(DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }) }, common_args: Cli { log_level: Some(Level(Debug)), no_cache: false, offline: false, files: CommonFileArgs { demo_files: ["demos/demos-v2.yaml"], stack_files: ["stacks/stacks-v2.yaml"], release_files: [] }, repos: CommonRepoArgs { helm_repo_stable: "https://repo.stackable.tech/repository/helm-stable/", helm_repo_test: "https://repo.stackable.tech/repository/helm-test/", helm_repo_dev: "https://repo.stackable.tech/repository/helm-dev/" }, subcommand: Demo(DemoArgs { subcommand: Install(DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }) }) }, cache: Cache { auto_purge_interval: 900s, backend: Disk { base_path: "/home/jim/.cache/stackablectl" }, max_age: 3600s }
    in run with self: Cli { log_level: Some(Level(Debug)), no_cache: false, offline: false, files: CommonFileArgs { demo_files: ["demos/demos-v2.yaml"], stack_files: ["stacks/stacks-v2.yaml"], release_files: [] }, repos: CommonRepoArgs { helm_repo_stable: "https://repo.stackable.tech/repository/helm-stable/", helm_repo_test: "https://repo.stackable.tech/repository/helm-test/", helm_repo_dev: "https://repo.stackable.tech/repository/helm-dev/" }, subcommand: Demo(DemoArgs { subcommand: Install(DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }) }) }

  ERROR  Go wrapper function go_add_helm_repo encountered an error: error converting JSON to YAML: yaml: control characters are not allowed
    at rust/stackable-cockpit/src/helm.rs:400
    in add_repo with repo_name: "vector", repo_url: "https://helm.vector.dev"
    in install_manifests
    in install_stack_manifests
    in install_cmd with args: DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }, common_args: Cli { log_level: Some(Level(Debug)), no_cache: false, offline: false, files: CommonFileArgs { demo_files: ["demos/demos-v2.yaml"], stack_files: ["stacks/stacks-v2.yaml"], release_files: [] }, repos: CommonRepoArgs { helm_repo_stable: "https://repo.stackable.tech/repository/helm-stable/", helm_repo_test: "https://repo.stackable.tech/repository/helm-test/", helm_repo_dev: "https://repo.stackable.tech/repository/helm-dev/" }, subcommand: Demo(DemoArgs { subcommand: Install(DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }) }) }, transfer_client: FileTransferClient { client: Client { accepts: Accepts, proxies: [Proxy(System({}), None)], referer: true, default_headers: {"accept": "*/*"} }, cache: Cache { auto_purge_interval: 900s, backend: Disk { base_path: "/home/jim/.cache/stackablectl" }, max_age: 3600s } }
    in run with self: DemoArgs { subcommand: Install(DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }) }, common_args: Cli { log_level: Some(Level(Debug)), no_cache: false, offline: false, files: CommonFileArgs { demo_files: ["demos/demos-v2.yaml"], stack_files: ["stacks/stacks-v2.yaml"], release_files: [] }, repos: CommonRepoArgs { helm_repo_stable: "https://repo.stackable.tech/repository/helm-stable/", helm_repo_test: "https://repo.stackable.tech/repository/helm-test/", helm_repo_dev: "https://repo.stackable.tech/repository/helm-dev/" }, subcommand: Demo(DemoArgs { subcommand: Install(DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }) }) }, cache: Cache { auto_purge_interval: 900s, backend: Disk { base_path: "/home/jim/.cache/stackablectl" }, max_age: 3600s }
    in run with self: Cli { log_level: Some(Level(Debug)), no_cache: false, offline: false, files: CommonFileArgs { demo_files: ["demos/demos-v2.yaml"], stack_files: ["stacks/stacks-v2.yaml"], release_files: [] }, repos: CommonRepoArgs { helm_repo_stable: "https://repo.stackable.tech/repository/helm-stable/", helm_repo_test: "https://repo.stackable.tech/repository/helm-test/", helm_repo_dev: "https://repo.stackable.tech/repository/helm-dev/" }, subcommand: Demo(DemoArgs { subcommand: Install(DemoInstallArgs { demo_name: "stackable-poc", skip_release: false, stack_parameters: [], parameters: [], local_cluster: CommonClusterArgs { cluster_type: None, cluster_name: "stackable-data-platform", cluster_nodes: 2, cluster_cp_nodes: 1 }, namespaces: CommonNamespaceArgs { operator_namespace: None, product_namespace: None } }) }) }

Error: demo command error

Caused by these errors (recent errors listed first):
  1: demo error
  2: stack error
  3: Helm error: failed to add Helm repo: error converting JSON to YAML: yaml: control characters are not allowed
  4: failed to add Helm repo: error converting JSON to YAML: yaml: control characters are not allowed
```

### Environment

_No response_

### Would you like to work on fixing this bug?

None

### PR DESCRIPTION
**Pull request by Techassi**
Fixes #131, implementation details are [here](https://github.com/stackabletech/stackable-cockpit/issues/131#issuecomment-1781148126).

### CODE REVIEW

**Comment by Techassi** [comment]
As suspected, this needs further work to make MacOS builds compile again.

**Comment by Techassi** [comment]
Compiles again!

File: rust/helm-sys/src/lib.rs [hunk 1]
```diff
@@ -1,37 +1,109 @@
...
+/// Checks if the result string is an error, and if so, returns the error message as a string.
+pub fn to_helm_error(result: &str) -> Option<String> {
+    if !result.is_empty() && result.starts_with(HELM_ERROR_PREFIX) {
+        return Some(result.replace(HELM_ERROR_PREFIX, ""));
+    }
+
+    None
+}
+
+unsafe fn cstr_ptr_to_string(c: *mut c_char) -> String {
```
Comment (sbernauer) [inline rust/helm-sys/src/lib.rs:103]: Could you please document that this function consumes the String by freeing it.

**Review by sbernauer — COMMENTED** [review]
Very nice! LGTM otherwise

File: rust/helm-sys/src/lib.rs (hunk 1 above)
Comment (Techassi) [reply rust/helm-sys/src/lib.rs:103]: Yes, I can and will!

Comment (Techassi) [reply rust/helm-sys/src/lib.rs:106]: Added in e00d8ec.

**Review by sbernauer — APPROVED** [review]
Thanks!

**Pull request merged**